lib/index.ts

```diff
@@ -109,11 +109,16 @@ export const operationalOops = (message: string, context?: {}) => {
     throw newOperationalOops(message, context)
 }
 
-export const newProgrammerOops = (message: string, context?: {}) => {
+export const newProgrammerOops = (
+    message: string,
+    context?: {},
+    error?: any
+) => {
     return new Oops({
         message,
-        category: constants.ProgrammerError,
+        category: getErrorCategory(error),
```

Comment:
or maybe handle it here that sometimes there is no error.
category: error ? getErrorCategory(error) || constants.ProgrammerError

Or change signature on getErrorCategory that error can be optional?

Reply:
I will change signature on getErrorCategory to show that error can be undefined